### PR TITLE
Search result: do not offer "Go to" if block height > chain tip

### LINKED ISSUE #4474: Don't offer "go to block" search result if you search for a block height > current block height
Searching for obviously non existing block should not present any result

image

The result leads to this page which is expected but unnecessary.

Screenshot 2023-12-04 at 1 20 01 PM

Instead, why not simply showing an appropriate string in the result area?

image

### PR DESCRIPTION
**Pull request by natsee**
Fixes #4474 

Screenshot 2023-12-13 at 19 24 25

### CODE REVIEW

File: frontend/src/app/components/search-form/search-results/search-results.component.html [hunk 1]
```diff
@@ -1,10 +1,13 @@
 <div class="dropdown-menu show" *ngIf="results" [hidden]="!results.hashQuickMatch && !results.addresses.length && !results.nodes.length && !results.channels.length">
-  <ng-template [ngIf]="results.blockHeight">
+  <ng-template [ngIf]="results.blockHeight && !results.blockHeightOutOfRange">
     <div class="card-title" i18n="search.bitcoin-block-height">Bitcoin Block Height</div>
     <button (click)="clickItem(0)" [class.active]="0 === activeIdx" type="button" role="option" class="dropdown-item">
       <ng-container *ngTemplateOutlet="goTo; context: { $implicit: results.searchText }"></ng-container>
     </button>
   </ng-template>
+  <ng-template #noBlockFound [ngIf]="results.blockHeight && results.blockHeightOutOfRange">
```
Comment (softsimon) [inline frontend/src/app/components/search-form/search-results/search-results.component.html:8]: We can remove this part.

We never show "Lightning Channel not found" or similar when posting wrong info ... We just display what it matches.

**Review by softsimon — CHANGES_REQUESTED** [review]
See comment

**Comment by natsee** [comment]
Removed the "No block found" message:

Screenshot 2023-12-14 at 09 42 20

**Review by softsimon — CHANGES_REQUESTED** [review]
I realized we don't need the `blockHeightOutOfRange` stuff anymore.. just adding this to the existing matchesBlockHeight check.


`const matchesBlockHeight = this.regexBlockheight.test(searchText) && parseInt(searchText) <= this.stateService.latestBlockHeight;`

**Review by softsimon — APPROVED** [review]
tested ACK @ [b0c02b16]

**Pull request merged**